### DISCUSSION TITLE
Vim syntax file for writing mtable parsers in ctags

### DISCUSSION THREAD

**Discussion by terminatorul** [universal-ctags/ctags#4017] [Show and tell]
I noticed there isn't anything like this when I was struggling with the regular expressions for my parser. so here is a Vim syntax file for mtable parsers for ctags:

https://www.vim.org/scripts/script.php?script_id=6112

Highlighting only for the default regular expressions (POSIX ERE extended expressions). Let me know if you find this usefull or if you found something similar that can help read parser file.

![image](https://github.com/universal-ctags/ctags/assets/378924/b5f26692-1ace-4cb5-8ce3-f0c767c9369d)

Contributions are welcome, there are many missing parts in the syntax:

- line regex and multi-line regex
- complete the syntax for ERE bracket expressions (character lists in gnulib)
  (some corner cases still exist)
- BREs (POSIX basic regular expressions)
- maybe pcre2, which I expect is more complicated
- some way to make Ghostscript work as a nested syntax (in Vim), or re-create it inline here

**Comment by masatake** [comment]
Though I maintained actively, I wrote a similar one for Emacs: https://github.com/universal-ctags/ctags/blob/master/misc/ctags-optlib-mode.el . 

You can put yours to https://github.com/universal-ctags/ctags/tree/master/misc if you want.

BTW, the name of script engine is optscript, not Ghostscript.

**Comment by shyun3** [comment]
Hi @terminatorul, would you consider putting your syntax file on GitHub? It would be more friendly to plugin managers that way.
